sql/src/main/java/io/crate/analyze/InsertFromSubQueryAnalyzedStatement.java

```diff
@@ -134,11 +146,20 @@ public InsertFromSubQueryAnalyzedStatement(AnalyzedRelation subQueryRelation,
                         symbol = InputColumns.create(generatedRef.generatedExpression(), sourceSymbols);
                     }
                 }
-                symbols.add(symbol);
+
                 if (SymbolVisitors.any(Symbols.IS_COLUMN, symbol)) {
-                    throw new IllegalArgumentException(String.format(Locale.ENGLISH,
-                        "Column \"%s\" is required but is missing from the insert statement", column.sqlFqn()));
+                    if (alwaysRequireColumn) {
+                        throw new IllegalArgumentException(String.format(
+                            Locale.ENGLISH,
+                            "Column \"%s\" is required but is missing from the insert statement", column.sqlFqn()));
+                    } else {
+                        symbols.add(Literal.NULL);
```

Comment:
i thought of having a separate methods for handling primary keys and partitioned by symbols, but they would essentially implement the same logic, except that if a partitioned column is missing we go for null and fail if the missing column is a primary key.  Therefore, i did it like that, not sure how satisfiable that is.